### PR TITLE
Split the doc view

### PR DESCRIPTION
**Pull request by tengqm**
This PR adds a splitter so that the left navigation panel can be resized.

### CODE REVIEW

**Review by dipesh-rawat — COMMENTED** [review]
The ability to resize the left sidebar could prove to be useful.

Looking at the changes in the [deploy preview](https://deploy-preview-41695--kubernetes-io-main-staging.netlify.app/docs/home/), ~it appears that I can't adjust the navbar size~ (its little difficult on Mac to get the resize option by hovering at the edge of the sidebar).

Additionally, the small screen mobile experience is broken, but that can be sorted with a few adjustments.

**_Current Mobile deploy render:_** 

mobile-preview

File: layouts/partials/head.html [hunk 1]
```diff
@@ -80,6 +80,8 @@
 {{ end }}
 
 <script src="/js/jquery-3.6.0.min.js" intregrity="sha384-vtXRMe3mGCbOeY7l30aIg8H9p3GdeSe4IFlP6G8JMa7o7lXvnz3GFKzPxzJdPfGK" crossorigin="anonymous"></script>
+<!--script src="https://unpkg.com/split.js/dist/split.min.js"></script-->
+<script src="/js/split-1.6.0.js"></script>
```
Comment (sftim) [inline layouts/partials/head.html:84]: We could set an `integrity` value too, even though it's local. Helps track when we made changes.

**Comment by tengqm** [comment]
> The ability to resize the left sidebar could prove to be useful.
> 
> Looking at the changes in the [deploy preview](https://deploy-preview-41695--kubernetes-io-main-staging.netlify.app/docs/home/), ~it appears that I can't adjust the navbar size~ (its little difficult on Mac to get the resize option by hovering at the edge of the sidebar).

To make the splitter visually attractive, some background or glyph are needed. We can do that later.

> Additionally, the small screen mobile experience is broken, but that can be sorted with a few adjustments.

Yup. Notice that.

**Comment by reylejano** [comment]
IMG_ADC75887E8CB-1-sm


current preview on mobile, the slider works but nothing renders in the left navigation pane

**Comment by tengqm** [comment]
@reylejano I'm aware of that defect and I've tried my best to solve it. However, it is more complicated than I expected. I'm not a Web developer anyway. Please suggest if this can be left for future improvement or we are not to add this partial solution.

**Comment by sftim** [comment]
Is there an issue that this fixes? I remember discussion about people not being happy with the way our flavor of Docsy renders things.

**Comment by tengqm** [comment]
@sftim This is an attempt to fix #39621. The problem is that we cannot resize the left navigation bar, where the text there gets wrapped, blurring the hierarchy of the content organization.

**Comment by kbhawkey** [comment]
👀

**Comment by sftim** [comment]
/lgtm
This does what it says it sets out to.

**Pull request merged**